### PR TITLE
Change qute:version git commit to display hash

### LINKED ISSUE #3095: Only show git commit (not git describe output) in version info

### PR DESCRIPTION
**Pull request by jgkamat**
Replaces output of git-describe

Closes #3095 

Here's a new output of `qute:version`

```
qutebrowser v1.0.1
Git commit: 07105f4fe34048d062a3ad384d8893ef1430c71f (2017-10-13 21:33:03 -0400)
Backend: QtWebEngine (Chromium 49.0.2623.111)
```

I really dislike `git-describe` and I saw the issue that's why I'm doing this :P 

Let me know if you intended a different format.

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3118)

### CODE REVIEW

**Comment by jgkamat** [comment]
I just updated the PR to use the method [described here](https://stackoverflow.com/questions/21017300/git-command-to-get-head-sha1-with-dirty-suffix-if-workspace-is-not-clean) rather than `rev-parse` directly. This gives us a `-dirty` at the end of commits while always giving us the hash. Thanks Enzime :)

File: qutebrowser/utils/version.py [hunk 1]
```diff
@@ -148,13 +148,18 @@ def _git_str_subprocess(gitpath):
     if not os.path.isdir(os.path.join(gitpath, ".git")):
         return None
     try:
-        cid = subprocess.check_output(
-            ['git', 'describe', '--tags', '--dirty', '--always'],
+        commit_hash = subprocess.check_output(
+            ['git',
+             'describe',
```
Comment (The-Compiler) [inline qutebrowser/utils/version.py]: nitpick: I think it's nicer to put those on a single line, as it's kind of a "sub-command"

File: qutebrowser/utils/version.py [hunk 2]
```diff
@@ -148,13 +148,18 @@ def _git_str_subprocess(gitpath):
...
+             '--match=NeVeRmAtCh',
```
Comment (The-Compiler) [inline qutebrowser/utils/version.py]: Please add a comment pointing to https://stackoverflow.com/a/21017394

File: qutebrowser/utils/version.py [hunk 3]
```diff
@@ -148,13 +148,18 @@ def _git_str_subprocess(gitpath):
...
+             '--always',
+             '--abbrev=40',
```
Comment (The-Compiler) [inline qutebrowser/utils/version.py]: I don't think we need the `--abbrev=40` - probably fine (and nicer) to show the short hash.

File: tests/unit/utils/test_version.py [hunk 4]
```diff
@@ -356,7 +356,9 @@ def _git(*args):
     def test_real_git(self, git_repo):
         """Test with a real git repository."""
         ret = version._git_str_subprocess(str(git_repo))
-        assert ret == 'foobar (1970-01-01 01:00:00 +0100)'
+        assert ret == \
+            '6e4b65a ' \
+            + '(1970-01-01 01:00:00 +0100)'
```
Comment (The-Compiler) [inline tests/unit/utils/test_version.py]: This probably fits on one line again now?

**Comment by The-Compiler** [comment]
Thanks!

**Pull request merged**